index.js

```diff
@@ -41,9 +41,24 @@ module.exports = function * (next) {
     return typeof item == 'number';
   });
 
+  //Adjust infinite end
+  if(end == Number.POSITIVE_INFINITY) {
+    if(this.length) {
+      end = this.length-1;
+    } else {
+      // This is ugly but we have to set something
+      end = 10000000000;
```

Comment:
Actually, it's always valid to ignore the range request so it can just be sent as a normal HTTP 200 response instead. I'll go for that.